### PR TITLE
[Enhancement] Exam Programming Exercises Show Template And Solution Participation

### PR DESCRIPTION
**Pull request by MDK5**
### Checklist
- [x] I tested *all* changes and *all* related features with different users (student, tutor, editor, instructor, admin) on the test server https://artemistest.ase.in.tum.de.
- [x] Client: I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client/).
- [x] Client: I documented the TypeScript code using JSDoc style.
- [x] Client: I added multiple screenshots/screencasts of my UI changes

### Motivation and Context
Normal programming exercises feature a link to the template and example submissions. Such link was missing for exam exercises.

### Description
Fixes the issue that prevented the links from getting shown and adds the right router link for exam exercises.

### Steps for Testing

1. Log in to Artemis
2. Create/Locate a normal programming exercise
3. Click the links for template and example submissions on the details page
4. Create/Locate an exam programming exercise
5. Click the links for template and example submissions on the details page
6. All links should work properly

### Test Coverage
not notably changed

### Screenshots
Before:
grafik
After:
grafik

### CODE REVIEW

File: src/main/webapp/app/exercises/programming/manage/programming-exercise-detail.component.ts [hunk 1]
```diff
@@ -288,4 +288,17 @@ export class ProgrammingExerciseDetailComponent implements OnInit, OnDestroy {
     private onError(error: HttpErrorResponse) {
         this.jhiAlertService.error(error.message);
     }
+
+    /**
+     * Generates the link to any participation's submissions, used for the link to template and solution submissions
+     * @param id id of the participation
```
Comment (MDK5) [inline src/main/webapp/app/exercises/programming/manage/programming-exercise-detail.component.ts:294]:
```suggestion
     * @param id of the participation
```

**Review by Malyuk-A — APPROVED** [review]
Tested on TS2 (during testing session). Links work for exam mode now. 👍
Code looks fine.

**Review by ndricimrr — APPROVED** [review]
Tested during testing session. Link working well 👍

**Review by iozbeyli — APPROVED** [review]
Tested during test session

**Pull request merged**